### PR TITLE
Extracting common mustache template

### PR DESCRIPTION
**Pull request by Riabkov**
- Update JavaScriptResource.mustache
- Cleanup JavaScriptBuilderElement
- Add missing properties to JavaScriptResource
- Add missing properties setup to JavaScriptBuilderElement
- Upgrade mustache compiler version to 0.9.11

- Catch PropertyMissingException in JavaScriptBuilderElement
- Add unit tests in JavaScriptBuilderTests
- Remove stored mustache template

- Add submodule with javascript template

### CODE REVIEW

File: pipeline.javascriptbuilder/src/main/java/fiftyone/pipeline/javascriptbuilder/templates/JavaScriptResource.java [hunk 1]
```diff
@@ -28,8 +28,11 @@
 public class JavaScriptResource {
     private final String _objName;
     private final String _jsonObject;
+    private final String _sessionId;
```
Comment (justadreamer) [inline pipeline.javascriptbuilder/src/main/java/fiftyone/pipeline/javascriptbuilder/templates/JavaScriptResource.java:31]: so we've added these parameters: sessionId, sequence, parameters, but
a) there are no tests covering this logic.  
b) there is no logic in `JavascriptBuilderElement` passing these here.

Let's look at dotnet implementation and also [spec](https://github.com/51Degrees/specifications/blob/36ff732360acb49221dc81237281264dac4eb897/pipeline-specification/pipeline-elements/javascript-builder.md) on how to implement these

File: pipeline.javascriptbuilder/src/main/java/fiftyone/pipeline/javascriptbuilder/Constants.java [hunk 2]
```diff
@@ -38,6 +38,6 @@ public class Constants {
     
     public static final String DEFAULT_PROTOCOL = "https";
     
-    public static final String TEMPLATE = "/fiftyone/pipeline/javascriptbuilder/templates/JavaScriptResource.mustache";
+    public static final String TEMPLATE = "/fiftyone/pipeline/javascriptbuilder/javascript-templates/JavaScriptResource.mustache";
```
Comment (justadreamer) [inline pipeline.javascriptbuilder/src/main/java/fiftyone/pipeline/javascriptbuilder/Constants.java:41]: do we need to rename the directory?  I think we can specify that the submodule checks out in the `template` directory, but that's cosmetical I guess

**Review by justadreamer — CHANGES_REQUESTED** [review]

**Comment by justadreamer** [comment]
Also please find a way to build a dev version of `pipeline-java` packages and to use them with `device-detection-java-examples` - so that the new Javascript template can be used with the cloud and onpremise examples.

File: pipeline.javascriptbuilder/src/test/java/fiftyone/pipeline/javascriptbuilder/JavaScriptBuilderTests.java [hunk 3]
```diff
@@ -72,146 +74,76 @@ public class JavaScriptBuilderTests {
...
         data.put("property", "thisIsAValue" );
-        
         doReturn(data).when(elementDataMock).asKeyMap();
+
+        int port = TcpHelper.getAvailablePort();
+        testServer = new TestServer(port);
+        testServer.start();
+
+        ChromeOptions options = new ChromeOptions();
+        options.setAcceptInsecureCerts(true);
```
[the remark this answers is not in the document]
Comment (Riabkov) [reply pipeline.javascriptbuilder/src/test/java/fiftyone/pipeline/javascriptbuilder/JavaScriptBuilderTests.java:119]: Added that from .NET just in case

**Review by justadreamer — APPROVED** [review]

**Pull request merged**